controllers/neutronapi_controller.go

```diff
@@ -720,6 +727,36 @@ func (r *NeutronAPIReconciler) reconcileNormal(ctx context.Context, instance *ne
 	instance.Status.Conditions.MarkTrue(condition.InputReadyCondition, condition.InputReadyMessage)
 	// run check OpenStack secret - end
 
+	memcached, err := r.getNeutronMemcached(ctx, helper, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.MemcachedReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				condition.MemcachedReadyWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, fmt.Errorf("memcached %s not found", instance.Spec.MemcachedInstance)
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.MemcachedReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.MemcachedReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	if !memcached.IsReady() {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.MemcachedReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.MemcachedReadyWaitingMessage))
+		return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, fmt.Errorf("memcached %s is not ready", memcached.Name)
```

Comment:
Overall the change itself is a good one and I don't have anything against it. 
I understand that we're going to introduce the dependency to a memcached instance, which means that the deployment won't progress if it's not present in the main ctlplane.
If we're happy with this config, then +1 from me, otherwise we need to improve the logic a bit and make it optional (e.g., if I don't specify a memcached instance to watch for, don't block the neutron operator deployment).

Reply:
and looking at TripleO this change seems consistent as we used to enable memcached by default as part of the regular overcloud roles [1]

[1] https://github.com/openstack/tripleo-heat-templates/blob/master/overcloud-resource-registry-puppet.j2.yaml#L163

Reply:
This is also consistent with at least heat and keystone operators: https://github.com/openstack-k8s-operators/keystone-operator/blob/main/controllers/keystoneapi_controller.go#L702 and https://github.com/openstack-k8s-operators/heat-operator/blob/main/controllers/heat_controller.go#L387 so I think we can/should keep it like it is proposed now. Any objections?

Reply:
Not from me, I'm +1

Reply:
Yes it would have been good to have this optional but as other base services depends on memcache already we can't avoid it fully. So optional part can be done later with other services if felt desired.